sympy/core/evalf.py

```diff
@@ -1391,6 +1391,31 @@ def evalf(self, n=15, subs=None, maxn=100, chop=False, strict=False, quad=None,
             verbose=<bool>
                 Print debug information (default=False)
 
+        Notes
+        =====
+
+        It is suggested to use expr.evalf(subs=...) over expr.subs(...).evalf()
+        or expr.evalf().subs(...) since expr.evalf(subs=...) avoids the loss of
+        significance caused due to naive substitution as the expression is run
+        through the evalf algorithm which takes into account, the various issues
+        that can cause loss of significance or precision. Naive substitution can
+        also include substitution of only a few rather than all symbols in the
+        expression.
+
+        Examples
+        ========
+
+        >>> import sympy
+        >>> from sympy.abc import x, y, z
+        >>> (x + y - z).subs({x: 1e100, y: 1, z: 1e100}) #case_1
```

Comment:
->suggest the following for the discussion

```
Notes
=====

When using Floats in a computation, precision errors may affect the 
final result when the Float values are substituted into an expression. 
The way to obtain an accurate result is to pass the desired values as 
the subs argument when doing the evaluation. For example, adding 
1e16 (a Float) to 1 will truncate to 1e16 and then subtracting 1e16 
will give 0 -- that is the result obtained from the following when trying 
to evaluate by simple substitution of values into the expression:

>>> values = {x: 1e16, y: 1, z: 1e16}
>>> (x + y - z).subs(values)
0

If evalf is used and the desired values are sent as the 
argument for subs, the correct value is obtained:

>>> (x + y - z).evalf(subs=values)
1

Using the subs argument for evalf is the accurate way to 
evaluate an expression at a desired set of values when 
precision errors may otherwise affect the calculation.
```

Reply:
Sure, I'll add this.

Reply:
I was suggesting replacement of what is there since it provides some additional specificity for why the subs argument should be used.